modules/geochemistry/src/utils/EquilibriumConstantInterpolator.C

```diff
@@ -0,0 +1,171 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EquilibriumConstantInterpolator.h"
+#include "MooseError.h"
+#include "MooseUtils.h"
+#include "libmesh/utility.h"
+
+EquilibriumConstantInterpolator::EquilibriumConstantInterpolator(
+    const std::vector<Real> & temperature, const std::vector<Real> & logk, const std::string type)
+  : LeastSquaresFitBase()
+{
+  // The number of temperature points and logk points must be equal
+  if (temperature.size() != logk.size())
+    mooseError("The temperature and logk data sets must be equal in length in "
+               "EquilibriumConstantInterpolator");
+
+  // A value of 500.0 in logk signifies 'no value', so should not be used in the fit
+  std::vector<Real> useful_temperature, useful_logk;
+  for (unsigned int i = 0; i < temperature.size(); ++i)
+    if (!MooseUtils::absoluteFuzzyEqual(logk[i], 500.0))
```

Comment:
Can the constructor contain this value of "no value" instead of hard-coding the 500.0? (Maybe default to 500.0.)  Or is 500.0 a universally-accepted geochemistry no-value?

Reply:
Good idea, I'll do this

Reply:
Sorry to be so annoying, @cpgr, but can this 500.0000 thing somehow be in the Reader too?